students/bplanica/lesson02/assignment/code/charges_calc.py

```diff
@@ -0,0 +1,114 @@
+"""
+Returns total price paid for individual rentals
+12/18/2019, BPA, added logging/debuging
+"""
+
+import argparse
+import json
+import datetime
+import math
+
+import sys
+import logging
+LOG_FORMAT = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+LOG_FILE = datetime.datetime.now().strftime("%Y-%m-%d") + '.log'
+
+FORMATTER = logging.Formatter(LOG_FORMAT)
+
+FILE_HANDLER = logging.FileHandler(LOG_FILE)
+FILE_HANDLER.setLevel(logging.WARNING)
+FILE_HANDLER.setFormatter(FORMATTER)
+
+CONSOLE_HANDLER = logging.StreamHandler()
+CONSOLE_HANDLER.setLevel(logging.DEBUG)
+CONSOLE_HANDLER.setFormatter(FORMATTER)
+
+LOGGER = logging.getLogger()
+LOGGER.setLevel(logging.DEBUG)
+LOGGER.addHandler(FILE_HANDLER)
+LOGGER.addHandler(CONSOLE_HANDLER)
+
+DEBUG_LEVEL = [logging.NOTSET, logging.ERROR, logging.WARNING, logging.DEBUG]
+
+def parse_cmd_arguments():
+    """parse command string arguments"""
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='debug enabled/level', default=0)
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
```

Comment:
This could be changed to this, addressing one specific exception type that's quite likely to occur:

```
def load_rentals_file(filename):
    """import data"""
    with open(filename) as file:
        try:
            data = json.load(file)
            logging.info("Data has been sucessfully imported")
        except json.decoder.JSONDecodeError as exc:
            logging.error(f"JSON source file contains formatting errors; Exception: {exc}")
            sys.exit()
    return data
```